tfx/components/util/tfxio_utils.py

```diff
@@ -23,13 +23,16 @@
 from tfx.types import artifact
 from tfx.types import standard_artifacts
 from tfx_bsl.tfxio import dataset_options
+from tfx_bsl.tfxio import parquet_tfxio
 from tfx_bsl.tfxio import raw_tf_record
 from tfx_bsl.tfxio import record_to_tensor_tfxio
 from tfx_bsl.tfxio import tf_example_record
 from tfx_bsl.tfxio import tf_sequence_example_record
 from tfx_bsl.tfxio import tfxio
 from tensorflow_metadata.proto.v0 import schema_pb2
 
+
+_SUPPORTED_PAYLOAD_FORMATS = ['parquet', 'tfrecords_gzip']
```

Comment:
nit: set or tuple

Reply:
consider to use [this](https://github.com/tensorflow/tfx/blob/master/tfx/proto/example_gen.proto#L123)

FORMAT_TFRECORDS_GZIP is currently supported, and FORMAT_PARQUET will be added.
ExampleGen will attach the file format in its output example artifact, which can be used by downstream to figure out the example artifact's file format

Reply:
ah nice, I missed that FileFormat proto message!

Reply:
@1025KB wondering if you also wanted me to change the type of `file_format` param in `make_tfxio` to `example_gen_pb2.FileFormat`?

Reply:
oh, I didn't notice it's Payload format, then using [this](https://github.com/tensorflow/tfx/blob/master/tfx/proto/example_gen.proto#L101) payload format is fine too

what I mean here is instead of hardcoded 'parquet', 'tfrecords_gzip' str, can we use enum in PayloadFormat proto here? otherwise I feel there will be a PayloadFormat.FORMAT_PARQUET to 'parquet' conversion somewhere?

iindyk@, any thoughts?

Reply:
done in https://github.com/tensorflow/tfx/pull/4761/commits/465ddff65590050c4a1adb74aeb7b01336af539c. The variable was actually badly named, it is indeed the supported file formats

Reply:
In general I think it makes sense to have enum for such thing instead of string, but if we want to update this we'd also need to update all callsites: components and tests, which is a bit trickier. If this proves to be messy (I think it may be) I'd keep it as is for now and add a todo to refactor this.

Reply:
I see, then a todo sound good

Reply:
@iindyk should i revert the last commit and add a TODO then?

Reply:
> should i revert the last commit and add a TODO then?
sgtm

> I did check the calls to make_tfxio within this project and non of them were passing the file format

Even tests? that's unfortunate this path is not covered. Aside from the upstream code we'd also need to update tfxios that are produced by the factory, bc they take strings (or make a bridge proto enum ->string).

For payload_format there are places that actually pass int (and I'm not yet sure if there's a good reason for that, this seems fragile).